exporter/containerimage/attestations.go

```diff
@@ -1,10 +1,169 @@
 package containerimage
 
 import (
+	"bytes"
+	"context"
+	"fmt"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	"github.com/moby/buildkit/cache"
+	"github.com/moby/buildkit/exporter/attestation"
+	gatewaypb "github.com/moby/buildkit/frontend/gateway/pb"
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/solver"
+	"github.com/moby/buildkit/solver/result"
 	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/spdx/tools-golang/jsonloader"
+	"github.com/spdx/tools-golang/jsonsaver"
+	"github.com/spdx/tools-golang/spdx"
 )
 
 var intotoPlatform ocispecs.Platform = ocispecs.Platform{
 	Architecture: "unknown",
 	OS:           "unknown",
 }
+
+// supplementSBOM modifies SPDX attestations to include the file layers
+func supplementSBOM(ctx context.Context, s session.Group, target cache.ImmutableRef, targetRemote *solver.Remote, refs map[string]cache.ImmutableRef, att result.Attestation) (result.Attestation, error) {
+	if att.Kind != gatewaypb.AttestationKindInToto {
+		return att, nil
+	}
+	if att.InToto.PredicateType != intoto.PredicateSPDX {
+		return att, nil
+	}
+
+	content, err := attestation.Read(ctx, s, refs, att)
+	if err != nil {
+		return att, err
+	}
+
+	doc, err := decodeSPDX(content)
+	if err != nil {
+		return att, err
+	}
+
+	layers := newFileLayerFinder(target, targetRemote)
+	modifyFile := func(f *spdx.File2_2) error {
+		_, desc, err := layers.find(ctx, s, f.FileName)
+		if err != nil {
+			return err
+		}
+		f.FileComment = fmt.Sprintf("layerID: %s", desc.Digest.String())
```

Comment:
What should we do if `f.FileComment` is already commented?

I'm not sure if there's a standard for composing comments together, so maybe this isn't the right sort of place to put this information?

Reply:
Have settled on not overwriting the `FileComment` and simply skipping over this file if there is already a comment - given that there's no way to safely detect what to do in this scenario. If some sort of standard emerges, we can try and follow that, but as of now, it doesn't seem like there is one.

Reply:
@wagoodman, should we use a different key here instead of `layerID`? Ideally we wouldn't want to interfere with an existing tooling that expects the value here to be a ChainID, so it might make sense to pick something different, like maybe just `layer`?